### PR TITLE
refactor new school vocab form out into its own component.

### PR DESCRIPTION
**Pull request by stopfstedt**
refs https://github.com/ilios/frontend/issues/6202
fixes https://github.com/ilios/frontend/issues/6226
image

image

### CODE REVIEW

**Review by dartajax — APPROVED** [review]
Nice.

**Comment by dartajax** [comment]
Once we get the @jrjohnson stamp of approval, it would be my honor to merge this.

File: app/components/school-vocabularies-expanded.js [hunk 1]
```diff
@@ -31,9 +31,9 @@ export default class SchoolVocabulariesExpandedComponent extends Component {
   }
 
   @restartableTask
-  *load(element, [school]) {
-    yield this.loadSchool(school.id);
-    const vocabularies = (yield school.vocabularies).toArray();
+  *load() {
```
Comment (jrjohnson) [inline app/components/school-vocabularies-expanded.js:34]: I don't see where this is triggered, am I missing it in the template?

File: app/components/school-vocabularies-list.hbs [hunk 2]
```diff
@@ -1,170 +1,104 @@
...
-                  {{t "general.terms"}}
-                </th>
-                <th class="text-left" colspan="1">
-                  {{t "general.actions"}}
-                </th>
+<div
+  class="school-vocabularies-list"
+  data-test-school-vocabularies-list
+  {{did-insert (perform this.load)}}
+  {{did-insert (perform this.load) @school.vocabularies}}
```
Comment (jrjohnson) [inline app/components/school-vocabularies-list.hbs:5]: This should be `did-update`, but would be even better replaced with something like https://github.com/ilios/frontend/blob/050667fbcdd103f9a1a189b910c144d366f40d28/app/components/program-year/competencies.js#L16

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: app/components/school-vocabularies-expanded.js (hunk 1 above)
Comment (stopfstedt) [reply app/components/school-vocabularies-expanded.js:34]: it's wired up in the corresponding template:
https://github.com/ilios/frontend/blob/master/app/components/school-vocabularies-expanded.hbs#L3

File: app/components/school-vocabularies-list.hbs (hunk 2 above)
Comment (stopfstedt) [reply app/components/school-vocabularies-list.hbs]: fixed :heavy_check_mark:

File: app/components/school-vocabularies-expanded.js (hunk 1 above)
Comment (jrjohnson) [reply app/components/school-vocabularies-expanded.js:34]: Ah, Github was hiding the top of that file. Sorry for the noise!

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**